### PR TITLE
parameter to add this plugin to custom war

### PR DESCRIPTION
**Pull request by vtitov**
Corresponding jira issue:
https://issues.jenkins-ci.org/browse/JENKINS-54949

### CODE REVIEW

File: src/main/java/org/jenkinsci/maven/plugins/hpi/WarMojo.java [hunk 1]
```diff
@@ -58,7 +67,18 @@ public void execute() throws MojoExecutionException {
             rezip.addZipfileset(z);
 
             getProject().setArtifacts(resolveDependencies(dependencyResolution));
-            for( MavenArtifact a : getProjectArtifacts() ) {
+
+            Set<MavenArtifact> projectArtifacts = new LinkedHashSet<>(getProjectArtifacts());
+            if(addThisPluginToCustomWar) {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/maven/plugins/hpi/WarMojo.java]: Would be nice to ensure that the artifact actually has the `hpi` packaging.

**Comment by oleg-nenashev** [comment]
LGTM, just some defensive code needed.
CC @raul-arabaolaza , because https://github.com/jenkinsci/custom-war-packager has a different default value for this case. I hit it in ci.jenkins.io-runner, and I will follow-up with a pull request

File: src/main/java/org/jenkinsci/maven/plugins/hpi/WarMojo.java (hunk 1 above)
Comment (vtitov) [reply src/main/java/org/jenkinsci/maven/plugins/hpi/WarMojo.java]: added packaging type check in commit [de594d9](https://github.com/jenkinsci/maven-hpi-plugin/pull/86/commits/de594d96b895b696a327dd231866d6159b48f9e0)

Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/maven/plugins/hpi/WarMojo.java]: thanks!

**Comment by oleg-nenashev** [comment]
@vtitov thanks a lot, will release it today

**Pull request merged**